x-pack/plugins/fleet/server/services/agents/agent_service.ts

```diff
@@ -119,6 +125,11 @@ class AgentClientImpl implements AgentClient {
     );
   }
 
+  public async getLatestAgentAvailableVersion(includeCurrentVersion = true) {
+    await this.#runPreflight();
+    return getLatestAvailableVersion(includeCurrentVersion);
+  }
```

Comment:
Good change here. Could you also update the usages internally to Fleet to use this?

Reply:
Addressed in [9724dfc7](https://github.com/elastic/kibana/pull/166811/commits/9724dfc780a11b71c541a4d55b3a27449a0ddbc6)